h/views/activity.py

```diff
@@ -88,17 +90,22 @@ def user_link(userid):
         }
 
 
-@view_defaults(route_name='activity.group_search',
-               renderer='h:templates/activity/search.html.jinja2')
+@view_defaults(route_name='group_read',
+               renderer='h:templates/activity/search.html.jinja2',
+               effective_principals=security.Authenticated,
+               has_feature_flag='search_page',
+               has_permission='read')
```

Comment:
I really like the clarity of this, to be honest. (Although later I would like to look into writing tests for the routing)